app/scripts/controllers/metametrics.js

```diff
@@ -605,6 +606,27 @@ export default class MetaMetricsController {
     return unique.length;
   }
 
+  /**
+   * @param {object} metamaskState
+   * @returns number of unique token addresses
+   */
+  _getNumberOfTokens(metamaskState) {
+    const allUniqueAddresses = Object.values(metamaskState.allTokens).reduce(
+      (result, tokensByAccountByChain) => {
+        Object.values(tokensByAccountByChain).forEach((tokensByAccount) => {
+          tokensByAccount.forEach((token) => {
+            result.add(token.address);
+          });
+        });
+
+        return result;
+      },
+      new Set(),
+    );
+
+    return allUniqueAddresses.size;
```

Comment:
Its not necessary to do a unique check here, The addresses represent contracts, even if a token manages to have the same address on many different networks (possible, I think, but difficult) the user would have different balances on each chain effectively making them different tokens. It's also possible for two different tokens to have the same address on different chains.

Reply:
😮 Thank you for the catch and for explaining this.  Maybe there is a guide I can read to help me better understand the structure of token contracts 

Updating the code to not do the unique filter here.

Reply:
@brad-decker updated here: https://github.com/MetaMask/metamask-extension/pull/14427/commits/0bd0772b8e3173b5593cf109e54377f703e993a1